fix(linter): get correct source offsets for astro files

camchenry opened the pull request:
Previously, we reported the `source_text` of an Astro file correctly, but the `start` offset was incorrectly set to before the `---` of the frontmatter. The start value now correctly points to the end of the split.

camchenry commented:
* **#6196** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6196?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @camchenry and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

DonIsaac commented on crates/oxc_linter/src/loader/partial_loader/astro.rs:
```diff
@@ -42,8 +42,9 @@ impl<'a> AstroPartialLoader<'a> {
-        let js_code =
-            Span::new(start + ASTRO_SPLIT.len() as u32, end).source_text(self.source_text);
+        // move start to the end of the ASTRO_SPLIT
+        let start = start + ASTRO_SPLIT.len() as u32;
```
This logic looks identical. Am I missing something?

camchenry replied:
yep, the issue here is that the `start` used for getting the source text had the length added to it, but then the `start` without the addition was stored as the start offset

camchenry committed:
fix(linter): get correct source offsets for astro files (#6196)

Previously, we reported the `source_text` of an Astro file correctly, but the `start` offset was incorrectly set to before the `---` of the frontmatter. The start value now correctly points to the end of the split.
